stacker/blueprints/variables/types.py

```diff
@@ -35,17 +45,37 @@ def create(self, value):
                 type
 
         """
-        if not self._many and len(value) > 1:
-            message = (
-                "Only one resource can be provided for single "
-                "TroposphereType"
-            )
-            raise ValueError(message)
-
-        output = [self._type.from_dict(title, values) for title, values in
-                  value.items()]
-
-        return output if self._many else output[0]
+        if self._optional and not value:
+            return None
+
+        if hasattr(self._type, 'resource_type'):
+            # Our type is a resource, so ensure we have a dict of title to
+            # parameters
+            if not isinstance(value, dict):
+                raise ValueError("Resources must be specified as a dict of "
+                                 "title to parameters")
+            if not self._many and len(value) > 1:
+                raise ValueError("Only one resource can be provided for this "
+                                 "TroposphereType variable")
+
+            result = [self._type.from_dict(title, v) for title, v in
+                      value.items()]
+        else:
+            # Our type is for properties, not a resource, so don't use
```

Comment:
Yes, I couldn't think of a better way to handle it. Having different types of input is confusing, but having them differ from what you would pass to the actual resource would be even more confusing IMO. I'll update the docs and try to explain things as best as I can.